internal/dcache/rpc/server/handler.go

```diff
@@ -1773,18 +1854,40 @@ func (h *ChunkServiceHandler) JoinMV(ctx context.Context, req *models.JoinMVRequ
 	mvInfo := rvInfo.getMVInfo(req.MV)
 	if mvInfo != nil {
 		//
-		// TODO: Till Sourav formally implements idempotent handling of JoinMV and UpdateMV RPCs,
-		//	 we have the following to not treat "double join" as failure.
-		//	 Double join can happen when let's say we have two or more outofsync component RVs
-		//	 for an MV and fixMV() sends JoinMV request to each of the outofsync RVs. If one or
-		//	 more of these fail, the joinMV() will treat it as a failure and not update clustermap.
-		//	 Next time when fixMV() is called it'll again attempt fixing and again send JoinMV.
-		//	 Note that for proper handling we need to ensure that the reservedSpace remains
-		//	 same across both calls. Also if an RV is joined but never used later (maybe joinMV()
-		//	 picked a new RV in the next iteration), we should time out and undo the reservedSpace.
+		// JoinMV and UpdateMV need to be idempotent to not treat "double join" as failure.
+		// Double join can happen when let's say we have two or more outofsync component RVs
+		// for an MV and fixMV() sends JoinMV request to each of the outofsync RVs. If one or
+		// more of these fail, the joinMV() will treat it as a failure and not update clustermap.
+		// Next time when fixMV() is called it'll again attempt fixing and again send JoinMV.
+		// Note that for proper handling we need to ensure that the reservedSpace remains
+		// same across both calls. Also if an RV is joined but never used later (maybe joinMV()
+		// picked a new RV in the next iteration), we should time out and undo the reservedSpace.
+		// This one is a TODO.
+		//
+		errStr := fmt.Sprintf("Double join for %s/%s, prev join at: %s, by: %s",
+			req.RVName, req.MV, mvInfo.joinedWhen, mvInfo.joinedBy)
```

Comment:
We can have asserts for `joinedWhen` and `joinedBy` when it is a multiple join scenario.
Also, should we mention "multiple join" instead of "double join" in the log?

Reply:
rvInfo.getMVInfo() has those asserts
"double join" is better as it emphasizes a join for something already joined.. multiple is not the key